### DISCUSSION TITLE
Support for Pokernow .json File Conversion?

### DISCUSSION THREAD

**Discussion by pmaster** [DrkYoda/PokerNow-OpenHandHistory-Converter--Python-#49] [General]
Are there any plans for supporting Pokernow replay files (.json) convertability?

Pokernow .csv files are obtained by downloading pokernow log files. Pokernow log files are limited to 20k hands and expire soon, while the .json replay files show full sessions and don't expire.

**Comment by DrkYoda** [comment]
I haven't considered this and was unaware that the .csv format is going to be discontinued.  I do recall when the hand replay was added that it was using .json to store the data, but it seemed semantically cryptic and was structured in a way that was unnecessarily complicated.  Are you asking if there are plans to convert old .csv hand histories to PokerNow .json files used by the hand replayer or if there are plans to convert PokerNow .json files to .ohh?

In either case the answer is no I don't have any plans to support the replay files, but if the csv format is discontinued then I will make plans to convert the PokerNow .json format to Open Hand Histories .ohh format.

**Reply by pmaster** [reply]
Just seeing this now.

As far as I know, PokerNow doesn't have plans to discontinue the .csv format. I was just noting that PokerNow doesn't store .csv files after several days, whereas they do store .json replay files allegedly forever (at least for premium members, not sure about otherwise).

I agree that the .json replay files are structured very poorly, unfortunately.

**Reply by pmaster** [reply]
Would you be open to potentially helping me convert 7 pokernow .csv files?

I spent several hours trying to convert them using your code, however kept running into bugs. I'm assuming the code works for you, and I was hoping you might be open to converting the files for me.

If so, is there a better place I should contact you? I have my email and telegram listed on my profile.

**Reply by pmaster** [reply]
I've continued trying to get this to work. After dealing with some random system bugs, then adjusting the regex to work (only) for PokerNow's setting in which there's only whole chips and no fractional chips (e.g., removing all instances of "\.\d{2}|\d+").

Now, I'm trying to fix ante handling. Currently, antes seem to be ignored.

**Reply by DrkYoda** [reply]
It does work and it converts the csv files very quickly but as you already discovered, the regular expressions will only see decimal numbers, also antes would not be seen.  I wrote this script with the intention that I would be the only person who would ever use so the script is built around the game settings I use.  However, I'd be happy to add those features, but I would need to have some csv hand histories from games that used integer amounts and antes, if you could send me your csv files it would be helpful.  The best way to contact me on discord, user name **shiner1976**, or you if you don't use discord then send me an email at sudduth.mark@gmail.com.

**Reply by DrkYoda** [reply]
Also, if you've already made the changes to the code, then submit a pull request and I'll merge it.  You do kind of sound like you might know what you're doing with programming, talking about regular expression and all.  Now that I think about it, if you know how to write code then when you looked at the script you probably thought "holy hell! was this guy high when he wrote this? It's just a deeply nested loop with no classes or functions.  He seems to have violated every best practices rule in a single script"  First of all, yes I was high when I wrote it.  Second...don't judge me man, this is a personal script not an API 😅

**Reply by pmaster** [reply]
Hahaha, looking forward to continuing the conversation on discord. Just added you (I'm pmaster).

**Comment by 0mn1c0de** [comment]
.json to ohh would be nice :) since that appears to be only way to download a history file now
